src/foam/dao/LinkPropertiesDAO.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'LinkPropertiesDAO',
+  extends: 'foam.dao.ProxyDAO',
+  flags: ['java'],
+
+  documentation: `
+    When you have two models with a relationship between them and you want a
```

Comment:
"with a relationship between them" - models are not using a foam.RELATIONSHIP correct? 
Can we use RELATIONSHIP as it specifies the relationship property and also dao keys, or is this overkill.

Reply:
We could probably do that. In the specific case of invoices though we'd need to update it to have a real relationship between `Invoice` and `User`, as right now it's just a `Reference` property on `Invoice`.

Reply:
Can you give a more concrete example of where/why we want this?

Can we achieve this by adding broader support to reference properties or create a new type of property?

Reply:
> Can you give a more concrete example of where/why we want this?

We have `Invoice` and `User`. `Invoice` has two Reference properties to `User`s: `payeeId` and `payerId`. In the TableView we display those properties as columns and we can use `tableCellFormatter` to do a lookup on the referenced `User` and display it's name, because the default `tableCellFormatter` is just the referenced `User`'s id, which is the value that the Reference property holds. This is all good until the user tries to search or sort by `payeeId` or `payerId`. Then it searches and sorts by the ids rather than the names that we're displaying.

One of the many possible solutions to this problem that we're considering is to add two dummy properties to `Invoice`: `payerName` and `payeeName` that will be of type String and hold the appropriate names. This way displaying is faster because we don't need to do the lookup every time, and sorting and searching works without needing extra UI support that requires more work on the user's part.

In order for those dummy properties to stay up to date, we'd need to decorate `localUserDAO` with a decorator that would update the appropriate `Invoice`s in `invoiceDAO` whenever a `User`'s name changes.

This PR implements that decorator, but generalized to any two models.

> Can we achieve this by adding broader support to reference properties or create a new type of property?

Maybe, that's one of the other solution areas we've thought about. If there were a way for MLangs on the server to do a lookup and check some property on another model then we could use that for the search and sort values rather than the id.